File: atcoder.jp/contests/abs/tasks/abc088_b/main.py
def input_int_list() -> "list[int]":
    return list(map(int, input().split()))


def main() -> None:
    _N = int(input())
    A = input_int_list()

    A.sort(reverse=True)
    alice = sum(A[::2])
    bob = sum(A[1::2])

    ans = alice - bob
    print(ans)


main()

File: atcoder.jp/contests/abs/tasks/abc088_b/main.hs
import Data.List (sort)

main :: IO ()
main = do
  n <- getInt
  a <- getIntList

  let sortedA = reverse . sort $ a
      alice = sum . step2 $ sortedA
      bob = sum . step2 . tail $ sortedA
      ans = alice - bob

  print ans

-- input functions

getInt :: IO Int
getInt = readLn

getIntList :: IO [Int]
getIntList = map read . words <$> getLine

-- functions

step2 :: [a] -> [a]
step2 [] = []
step2 [x] = [x]
step2 (x : y : xs) = x : step2 xs
